shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -51,10 +51,48 @@ bool GeometryComparator(SemanticsObject* a, SemanticsObject* b) {
 
 }  // namespace
 
+/**
+ * Represents a semantics object that has children and hence has to be presented to the OS as an
+ * UIAccessibilityContainer.
+ *
+ * The SemanticsObject class cannot implement the UIAccessibilityContainer protocol because an
+ * object that returns YES for isAccessibilityElement cannot also implement
+ * UIAccessibilityContainer.
+ *
+ * With the help of SemanticsObjectContainer, the hierarchy of semantic objects received from
+ * the framework...
+ *
+ * SemanticsObject1
+ *     SemanticsObject2
+ *         SemanticsObject3
+ *         SemanticsObject4
+ *
+ * ...is translated into the following hierarchy, which is understood by iOS:
+ *
+ * SemanticsObjectContainer1
+ *     SemanticsObject1
+ *     SemanticsObjectContainer2
+ *         SemanticsObject2
+ *         SemanticsObject3
+ *         SemanticsObject4
+ *
+ * The SemanticsObjects have semantic information attached to them which is interpreted by
+ * VoiceOver (they return YES for isAccessibilityElement). The SemanticsObjectContainers are just
+ * there for structure and they don't provide any semantic information to VoiceOver (they return
+ * NO for isAccessibilityElement).
```

Comment:
It might be worth adding a little more detail on how the transformation between the first tree (Flutter's view of the world) to second tree (iOS's view of the world). Specifically that we construct a sort of shadow-tree of our own tree, where each node in the iOS view of the world is now a `UIAccessibilityContainer` whose children are (`0`) the original `SemanticsObject` and (`1..n`) child UIAccessibilityContainers.